src/app/components/challenge/challengesettings/challengesettings.component.ts

```diff
@@ -118,11 +120,22 @@ export class ChallengesettingsComponent implements OnInit {
     return String(email).search (regex) !== -1;
   }
 
+  reflectChange() {
+    if (this.bannedEmailIds.toString() === this.formerBannedEmailIds.toString()) {
+      this.globalService.showToast('error', 'No change to reflect!');
+    } else if (this.isValidationError) {
+      this.globalService.showToast('error', 'Please enter a valid email!');
+    } else {
+      this.updateBannedEmailList();
+    }
+  }
+
   updateBannedEmailList() {
     const SELF = this;
     const BODY = JSON.stringify({
       banned_email_ids: SELF.bannedEmailIds
     });
+    this.formerBannedEmailIds = this.bannedEmailIds.concat(); // Creating deep copy
```

Comment:
We should update this variable after the success of API call.

Reply:
You are definitely right! Fixed :+1: